It seems there was a broken commit meant to fix compilation on win32,
but this broke compilation on musl.

See:

https://sourceforge.net/p/iperf2/code/ci/c8f861a0a07b0606a581f2658c71959ad9417923/

for the broken commit. The current patch fixes that by completing the
definition of gettcpinfo() to what it was before the broken commit.

--- a/compat/gettcpinfo.c	2021-12-05 17:49:56.000000000 -0300
+++ b/compat/gettcpinfo.c	2021-12-11 19:29:24.850947483 -0300
@@ -99,4 +99,7 @@
 };
 #else
 inline void gettcpinfo (int sock, struct ReportStruct *sample) {
+    sample->tcpstats.rtt = 1;
+    sample->tcpstats.isValid  = false;
+};
 #endif
